dj/api/data.py

```diff
@@ -83,3 +87,46 @@ def add_availability(
         status_code=200,
         content={"message": "Availability state successfully posted"},
     )
+
+
+@router.get("/data/{node_name}/")
+def data_for_node(
+    node_name: str,
+    *,
+    dimensions: List[str] = Query([]),
+    filters: List[str] = Query([]),
+    async_: bool = False,
+    session: Session = Depends(get_session),
+    query_service_client: QueryServiceClient = Depends(get_query_service_client),
+) -> QueryWithResults:
+    """
+    Gets data for a node
+    """
+    node = get_node_by_name(session, node_name)
+    if node.type not in (NodeType.METRIC, NodeType.CUBE, NodeType.DIMENSION):
+        raise DJException(message=f"Can't get data for node type {node.type}!")
+
+    if node.type == NodeType.DIMENSION:
+        if dimensions or filters:
```

Comment:
Why not for Dimension node data with filters?

Reply:
Good point, in thinking about this more, we can open this up to filters.

Reply:
Done! I've added support for dimensions with filters.